docs/adr/0001-mobile-app-architecture.md

```diff
@@ -0,0 +1,107 @@
+
+### Positive Consequences <!-- optional -->
+
+- [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+- …
+
+### Negative Consequences <!-- optional -->
+
+- [e.g., compromising quality attribute, follow-up decisions required, …]
+- …
+
+## Pros and Cons of the Options <!-- optional -->
+
+### 2 mobile apps in Swift (iOS) and Kotlin (Android)
+
+Build and maintain 2 distinct code-bases and repos.
+Example: [Signal](https://github.com/signalapp/)
+
+- ❇️ Good, because the apps are in the app stores.
+- ❇️ Good, because each platform can have the "purest" experience possible.
+- ❇️ Good, because devs will build with languages and tooling that are optimized for each
+  platform.
+- ⛔️ Bad, because we have to maintain 2 separate code-bases.
+- ⛔️ Bad, because we have to train ENGRs on 2 new tech stacks.
+- ⛔️ Bad, because each code-base needs at least 2 ENGRs for it; likely we need to dedicate
+  4 ENGRs.
+- ⛔️ Bad, because we have to build 2 new release pipelines.
+
+### Re-use C++/Qt code-base from VPN
+
+Re-use the VPN C++/Qt code-base and build the code into "Relay mode" to make a Relay
+app.
+Example: [Mozilla VPN](https://github.com/mozilla-mobile/mozilla-vpn-client)
+
+- ❇️ Good, because the apps are in the app stores.
+- ❇️ Good, because it leverages existing mobile app and release pipeline.
+- ❇️ Good, because it's a single code-base for both iOS and Android.
+- ⚠️ Bad, because we have to train ENGRs on new tech stack.
+- ⚠️ Bad, because the C++/Qt code-base needs significant work to do this.
+
+### Mobile Web
+
+Deliver Relay Phone app UX via a mobile website.
+Example: https://voice.google.com
+
+- ❇️ Good, because it leverages existing web app and release pipeline.
+- ❇️ Good, because it's a single code-base for both iOS and Android.
+- ❇️ Good, because ENGRs already know the tech stack.
+- ⚠️ Bad, because the Relay phone app isn't in the app stores.
+
+### React Native
+
+Build a React Native app.
+Example: [Joplin](https://github.com/laurent22/joplin/tree/dev/packages/app-mobile)
+Example: [Relay prototype](https://github.com/codemist/fx-private-relay-mobile-app)
+
+- ❇️ Good, because the apps are in the app stores.
+- ❇️ Good, because it's a single code-base for both iOS and Android.
+- ⚠️ Bad, because ENGRs need to learn a new tech stack.
+- ⚠️ Bad, because we have to build a new release pipeline.
+
+### React Native WebView
+
+Build a React Native app that uses a `WebView` to embed the mobile app.
+Example: [Relay prototype](https://github.com/Vinnl/fx-private-relay-webapp-wrapper/)
+
+- ❇️ Good, because the apps are in the app stores.
+- ❇️ Good, because it's a single code-base for both iOS and Android.
+- ❇️ Good, because it's probably the quickest-to-market option.
+- ❇️⚠️ Good-ish, because ENGRs know more of the tech stack.
+- ❇️⚠️ Good-ish, because it mostly leverages existing web app and release pipeline.
```

Comment:
```suggestion
- ❇️ Good, because it's probably the quickest-to-market option.
- ❇️ Good, because the web app can be augmented with native functionality where needed.
- ❇️ Good, because it can be incrementally transitioned to a full React Native app if conditions change.
- ❇️ Good, because many updates can be instantly activated for all users without having to go through app store reviews, making it easier to time releases.
- ❇️⚠️ Good-ish, because ENGRs know more of the tech stack.
- ❇️⚠️ Good-ish, because it mostly leverages existing web app and release pipeline.
- ⚠️❇️ Bad-ish, because we might need to bend over backwards to add native functionality on first release to satisfy app store reviewers that it's not just a repackaged website.
- ⚠️ Bad, because changes to the website can also affect the mobile experience, increasing the testing matrix for website changes.
```

Reply:
Links regarding app store issues with WebView: 
- https://daringfireball.net/2023/01/ice_cubes_app_store_limbo
- https://developer.apple.com/forums/thread/688288